src/System.Drawing.Common/tests/mono/System.Drawing.Imaging/IconCodecTests.cs

```diff
@@ -0,0 +1,2131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+//
+// ICO Codec class testing unit
+//
+// Authors:
+// 	Jordi Mas i Hernàndez (jordi@ximian.com)
+//	Sebastien Pouliot  <sebastien@ximian.com>
+//
+// Copyright (C) 2006-2007 Novell, Inc (http://www.novell.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Security.Permissions;
+using System.Text;
+using Xunit;
+
+namespace MonoTests.System.Drawing.Imaging
+{
+
+    public class IconCodecTest
+    {
+
+        /* Get suffix to add to the filename */
+        internal string GetOutSufix()
+        {
+            string s;
+
+            int p = (int)Environment.OSVersion.Platform;
+            if ((p == 4) || (p == 128) || (p == 6))
```

Comment:
Use `PlatformDetection.IsWindows`. 

Btw I see this method been used in multiple (or all) the test classes you're adding. Can we move it to a helper method?